### PR TITLE
MTM-44093 mqtt token refresh (Generic MQTT)

### PR DESCRIPTION
**Pull request by andrekramer1**

### CODE REVIEW

File: java-client/src/test/java/com/cumulocity/sdk/client/messaging/notifications/TokenApiImplTest.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public void setup() throws Exception {
     public void shouldCreateToken() {
         //given
         NotificationTokenRequestRepresentation tokenRequest =
-                new NotificationTokenRequestRepresentation("sub", "sup", 1L, false);
+                new NotificationTokenRequestRepresentation(null, "sub", "sup", true, 1L, false);
```
Comment (gapa-c8y) [inline java-client/src/test/java/com/cumulocity/sdk/client/messaging/notifications/TokenApiImplTest.java]: I might be getting mixed up, but should the null value be 3rd (type) parameter here?

File: java-client/src/test/java/com/cumulocity/sdk/client/messaging/notifications/TokenApiImplTest.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public void shouldCreateToken() {
     public void shouldBuildCreateTokenUri() {
         //given
         NotificationTokenRequestRepresentation tokenRequest =
-                new NotificationTokenRequestRepresentation("sub", "sup", 1L, false);
+                new NotificationTokenRequestRepresentation(null, "sub", "sup", true, 1L, false);
```
Comment (gapa-c8y) [inline java-client/src/test/java/com/cumulocity/sdk/client/messaging/notifications/TokenApiImplTest.java]: Same here (should null be 3rd param)

**Review by gapa-c8y — CHANGES_REQUESTED** [review]
Might be mixing up params in my head, so either correct me or the PR!

File: java-client/src/test/java/com/cumulocity/sdk/client/messaging/notifications/TokenApiImplTest.java (hunk 1 above)
Comment (andrekramer1) [reply java-client/src/test/java/com/cumulocity/sdk/client/messaging/notifications/TokenApiImplTest.java]: Good spot! thanks.

**Review by gapa-c8y — DISMISSED** [review]

**Review by rohitjoshi-c8y — DISMISSED** [review]

**Review by rohitjoshi-c8y — APPROVED** [review]

**Review by gapa-c8y — APPROVED** [review]

**Review by bggov — APPROVED** [review]

**Pull request merged**